metadata_layer.py

```diff
@@ -950,29 +957,19 @@ def apply(self, _db, collection, replace=None):
             new_lpdms.append(lpdm)
 
         if replace.formats:
-            # If any LicensePoolDeliveryMechanisms were not mentioned in
-            # self.formats, remove them.
+            # If any preexisting LicensePoolDeliveryMechanisms were
+            # not mentioned in self.formats, remove the corresponding
+            # LicensePoolDeliveryMechanisms.
             for lpdm in old_lpdms:
                 if lpdm not in new_lpdms:
                     for loan in lpdm.fulfills:
                         self.log.info("Loan %i is associated with a format that is no longer available. Deleting its delivery mechanism." % loan.id)
                         loan.fulfillment = None
-                    _db.delete(lpdm)
+                    # This can cause an open-access LicensePool to go
+                    # non-open-access.
+                    lpdm.delete()
 
-        # Changes to the delivery mechanisms may have changed the
-        # open-access status of all LicensePools for this work.
```

Comment:
This is now handled by `LicensePoolDeliveryMechanism.set()` and `LicensePoolDeliveryMechanism.delete()`